examples/demo_threading/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+                            event_sender
+                                .try_send(NetworkDataArrived::Disconnect(uuid))
+                                .unwrap();
+
+                            Response {
+                                status: Status::Ok,
+                                uuid: Some(uuid),
+                            }
+                        } else {
+                            Response {
+                                status: Status::ErrorNoUuid,
+                                uuid: None,
+                            }
+                        }
+                    }
+                    RequestCommand::Power { value } => {
+                        if let Some(uuid) = request.uuid {
+                            // Validate that our power is within the expected range
+                            if value < 0.0 || value > 1000.0 {
+                                Response {
+                                    status: Status::ErrorInvalidPower,
+                                    uuid: None,
+                                }
+                            } else {
+                                event_sender
+                                    .try_send(NetworkDataArrived::Power(uuid, value))
+                                    .unwrap();
+
+                                Response {
+                                    status: Status::Ok,
+                                    uuid: Some(uuid),
+                                }
+                            }
+                        } else {
+                            Response {
+                                status: Status::ErrorNoUuid,
+                                uuid: None,
+                            }
+                        }
+                    }
+                },
+                Err(_) => Response {
+                    status: Status::Error,
+                    uuid: None,
+                },
+            };
+
+            stream
+                .write(serde_json::to_string(&response).unwrap().as_bytes())
+                .await
+                .unwrap();
+            stream.flush().await.unwrap();
+        }
+
+        #[invokable]
+        fn start_server(&mut self, cpp: &mut CppObj) {
+            if self.join_handle_network.is_some() || self.join_handle_processing.is_some() {
+                println!("Already running a thread!");
+                return;
+            }
+
+            let (network_sender, mut network_receiver) = channel(4096);
+            let mut qt_sender = self.qt_sender.clone();
+            let update_requester = cpp.update_requester();
+
+            // Prepare our processing thread which builds average, count, total
+            let run_processing = async move {
+                let mut sensors: HashMap<Uuid, f64> = HashMap::new();
+
+                loop {
+                    let mut changed = false;
+
+                    Delay::new(Duration::from_millis(8)).await;
```

Comment:
This causes CPU usage, instead using a blocking channel where recv() waits or CondVar might be a better solution.